src/display_device/display_device.h

```diff
@@ -0,0 +1,313 @@
+    inactive,
+    active,
+    primary /**< Primary state is also implicitly active. */
+  };
+
+  /**
+   * @brief The device's HDR state in the operating system.
+   */
+  enum class hdr_state_e {
+    unknown, /**< HDR state could not be retrieved from the OS (even if the display supports it). */
+    disabled,
+    enabled
+  };
+
+  // For JSON serialization for hdr_state_e
+  NLOHMANN_JSON_SERIALIZE_ENUM(hdr_state_e, { { hdr_state_e::unknown, "unknown" },
+                                              { hdr_state_e::disabled, "disabled" },
+                                              { hdr_state_e::enabled, "enabled" } })
+
+  /**
+   * @brief Ordered map of [DEVICE_ID -> hdr_state_e].
+   */
+  using hdr_state_map_t = std::map<std::string, hdr_state_e>;
+
+  /**
+   * @brief The device's HDR state in the operating system.
+   */
+  struct device_info_t {
+    std::string display_name; /**< A name representing the OS display (source) the device is connected to. */
+    std::string friendly_name; /**< A human-readable name for the device. */
+    device_state_e device_state; /**< Device's state. @see device_state_e */
+    hdr_state_e hdr_state; /**< Device's HDR state. @see hdr_state_e */
+  };
+
+  /**
+   * @brief Ordered map of [DEVICE_ID -> device_info_t].
+   * @see device_info_t
+   */
+  using device_info_map_t = std::map<std::string, device_info_t>;
+
+  /**
+   * @brief Display's resolution.
+   */
+  struct resolution_t {
+    unsigned int width;
+    unsigned int height;
+
+    // For JSON serialization
+    NLOHMANN_DEFINE_TYPE_INTRUSIVE(resolution_t, width, height)
+  };
+
+  /**
+   * @brief Display's refresh rate.
+   * @note Floating point is stored in a "numerator/denominator" form.
+   */
+  struct refresh_rate_t {
+    unsigned int numerator;
+    unsigned int denominator;
+
+    // For JSON serialization
+    NLOHMANN_DEFINE_TYPE_INTRUSIVE(refresh_rate_t, numerator, denominator)
+  };
+
+  /**
+   * @brief Display's mode (resolution + refresh rate).
+   * @see resolution_t
+   * @see refresh_rate_t
+   */
+  struct display_mode_t {
+    resolution_t resolution;
+    refresh_rate_t refresh_rate;
+
+    // For JSON serialization
+    NLOHMANN_DEFINE_TYPE_INTRUSIVE(display_mode_t, resolution, refresh_rate)
+  };
+
+  /**
+   * @brief Ordered map of [DEVICE_ID -> display_mode_t].
+   * @see display_mode_t
+   */
+  using device_display_mode_map_t = std::map<std::string, display_mode_t>;
+
+  /**
+   * @brief A LIST[LIST[DEVICE_ID]] structure which represents an active topology.
+   * @note On Windows the order does not matter of both device ids or the inner lists.
+   * 
+   * EXAMPLES:
+   * ```cpp
+   * Single display:
+   *     [[DISPLAY_1]]
+   * 2 extended displays:
+   *     [[DISPLAY_1], [DISPLAY_2]]
+   * 2 duplicated displays:
+   *     [[DISPLAY_1, DISPLAY_2]]
+   * Mixed displays:
+   *     [[EXTENDED_DISPLAY_1], [DUPLICATED_DISPLAY_1, DUPLICATED_DISPLAY_2], [EXTENDED_DISPLAY_2]]
+   * ```
```

Comment:
The example should show how to use the code... this looks like example results of the code.